File: windows/Scanner/Models/Severity.cs
namespace AIExposureScanner.Scanner.Models;

public enum Severity
{
    Critical,
    High,
    Medium,
    Low
}

public static class SeverityExtensions
{
    public static string ToJsonValue(this Severity severity) =>
        severity switch
        {
            Severity.Critical => "critical",
            Severity.High => "high",
            Severity.Medium => "medium",
            Severity.Low => "low",
            _ => throw new ArgumentOutOfRangeException(nameof(severity), severity, null)
        };

    public static int SortRank(this Severity severity) =>
        severity switch
        {
            Severity.Critical => 0,
            Severity.High => 1,
            Severity.Medium => 2,
            Severity.Low => 3,
            _ => throw new ArgumentOutOfRangeException(nameof(severity), severity, null)
        };
}

File: macos/Scanner/Sources/Scanner/Models/Severity.swift
public enum Severity: String, Codable, Sendable {
    case critical
    case high
    case medium
    case low

    var sortRank: Int {
        switch self {
        case .critical:
            0
        case .high:
            1
        case .medium:
            2
        case .low:
            3
        }
    }
}
